File: java/lib/src/main/java/com/svix/api/Stream.java
// this file is @generated
package com.svix.api;

import com.svix.SvixHttpClient;
import com.svix.exceptions.ApiException;
import com.svix.models.EndpointHeadersOut;
import com.svix.models.HttpSinkHeadersPatchIn;
import com.svix.models.SinkTransformationOut;

import lombok.Getter;

import okhttp3.HttpUrl;

import java.io.IOException;

public class Stream {
    private final SvixHttpClient client;

    @Getter private final StreamEventType eventType;
    @Getter private final StreamEvents events;
    @Getter private final StreamSink sink;
    @Getter private final StreamStream stream;

    public Stream(SvixHttpClient client) {
        this.client = client;
        this.eventType = new StreamEventType(client);
        this.events = new StreamEvents(client);
        this.sink = new StreamSink(client);
        this.stream = new StreamStream(client);
    }

    /** Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks. */
    public EndpointHeadersOut sinkHeadersGet(final String streamId, final String sinkId)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/stream/%s/sink/%s/headers", streamId, sinkId));
        return this.client.executeRequest("GET", url.build(), null, null, EndpointHeadersOut.class);
    }

    /** Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks. */
    public EndpointHeadersOut sinkHeadersPatch(
            final String streamId,
            final String sinkId,
            final HttpSinkHeadersPatchIn httpSinkHeadersPatchIn)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/stream/%s/sink/%s/headers", streamId, sinkId));
        return this.client.executeRequest(
                "PATCH", url.build(), null, httpSinkHeadersPatchIn, EndpointHeadersOut.class);
    }

    /** Get the transformation code associated with this sink. */
    public SinkTransformationOut sinkTransformationGet(final String streamId, final String sinkId)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/stream/%s/sink/%s/transformation",
                                        streamId, sinkId));
        return this.client.executeRequest(
                "GET", url.build(), null, null, SinkTransformationOut.class);
    }
}

File: kotlin/lib/src/main/kotlin/Stream.kt
// this file is @generated
package com.svix.kotlin

import com.svix.kotlin.models.EndpointHeadersOut
import com.svix.kotlin.models.HttpSinkHeadersPatchIn
import com.svix.kotlin.models.SinkTransformationOut

class Stream(private val client: SvixHttpClient) {
    val eventType: StreamEventType = StreamEventType(client)

    val events: StreamEvents = StreamEvents(client)

    val sink: StreamSink = StreamSink(client)

    val stream: StreamStream = StreamStream(client)

    /** Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks. */
    suspend fun sinkHeadersGet(streamId: String, sinkId: String): EndpointHeadersOut {
        val url =
            client.newUrlBuilder().encodedPath("/api/v1/stream/$streamId/sink/$sinkId/headers")
        return client.executeRequest<Any, EndpointHeadersOut>("GET", url.build())
    }

    /** Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks. */
    suspend fun sinkHeadersPatch(
        streamId: String,
        sinkId: String,
        httpSinkHeadersPatchIn: HttpSinkHeadersPatchIn,
    ): EndpointHeadersOut {
        val url =
            client.newUrlBuilder().encodedPath("/api/v1/stream/$streamId/sink/$sinkId/headers")

        return client.executeRequest<HttpSinkHeadersPatchIn, EndpointHeadersOut>(
            "PATCH",
            url.build(),
            reqBody = httpSinkHeadersPatchIn,
        )
    }

    /** Get the transformation code associated with this sink. */
    suspend fun sinkTransformationGet(streamId: String, sinkId: String): SinkTransformationOut {
        val url =
            client
                .newUrlBuilder()
                .encodedPath("/api/v1/stream/$streamId/sink/$sinkId/transformation")
        return client.executeRequest<Any, SinkTransformationOut>("GET", url.build())
    }
}
